components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/logout/idpinit/processor/FederatedIdpInitLogoutProcessor.java

```diff
@@ -0,0 +1,635 @@
+
+        String clientId = null;
+        // Get the client id from the authenticator config.
+        for (Property property : idp.getDefaultAuthenticatorConfig().getProperties()) {
+            String propertyName = property.getName();
+            if (propertyName.equals(OIDCAuthenticatorConstants.IdPConfParams.CLIENT_ID)) {
+                clientId = property.getValue();
+                break;
+            }
+        }
+        // Check whether the client id exist in the aud claim.
+        if (StringUtils.isNotBlank(clientId)) {
+            if (!aud.contains(clientId)) {
+                throw new LogoutClientException(
+                        String.format(ErrorMessages.LOGOUT_TOKEN_AUD_CLAIM_VALIDATION_FAILED.getCode(), clientId),
+                        ErrorMessages.LOGOUT_TOKEN_AUD_CLAIM_VALIDATION_FAILED.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Do the iat claim validation according to OIDC back-channel logout specification
+     * Read the authenticator configs to check whether the iat validation is enabled and if enabled the get the
+     * validity period.
+     *
+     * @param iat
+     * @return boolean if validation is successful.
+     * @throws LogoutClientException
+     */
+    private void validateIat(Date iat) throws LogoutClientException {
+
+        if (iat == null) {
+            throw handleLogoutClientException(ErrorMessages.LOGOUT_TOKEN_IAT_VALIDATION_FAILED);
+        }
+        if (Boolean.parseBoolean(getAuthenticatorConfig().getParameterMap()
+                .get(OIDCAuthenticatorConstants.BackchannelLogout.ENABLE_IAT_VALIDATION))) {
+            // iatValidityPeriod will be in seconds.
+            long iatValidityPeriod = getIatValidityPeriod();
+            long issuedAtTimeMillis = iat.getTime();
+            long currentTimeInMillis = System.currentTimeMillis();
+            long iatValidityPeriodInMillis = iatValidityPeriod * 1000;
+
+            if (currentTimeInMillis - issuedAtTimeMillis > iatValidityPeriodInMillis) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Logout token is used after iat validity period." +
+                            " iat validity period(m): " + iatValidityPeriod +
+                            ", Current Time : " + currentTimeInMillis +
+                            ". This logout token is not valid.");
+                }
+                throw handleLogoutClientException(ErrorMessages.LOGOUT_TOKEN_IAT_VALIDATION_FAILED);
+            }
+            if (log.isDebugEnabled()) {
+                log.debug("iat validity period of logout token was validated successfully.");
+            }
+        }
+    }
+
+    /**
+     * Get iatValidityPeriod from configuration file.
+     * Use default value if error occurs.
+     *
+     * @return - iatValidityPeriod.
+     */
+    private long getIatValidityPeriod() {
+
+        long iatValidityPeriod = 0;
```

Comment:
is this variable required? We can directly return the value instead of assigning it to this variable.